index.js

```diff
@@ -91,47 +91,69 @@ OpenDKIMVerifyStream.prototype._build_result = function(error) {
   return result;
 };
 
-OpenDKIMVerifyStream.prototype.write = function(buf) {
-  if (buf && buf.length && !(this.finished)) {
-    var chunk = this._get_chunk(buf);
-    try {
-      this.opendkim.chunk({
+OpenDKIMVerifyStream.prototype.write = function (buf) {
+  var self = this;
+
+  if (buf && buf.length && !(self.finished)) {
+    var chunk = self._get_chunk(buf);
+    var options = {
         message: chunk,
         length: chunk.length
-      });
-    } catch (err) {
-      return this.callback(err, this._build_result(err));
-    }
+    };
+    self.opendkim.chunk(options, function (err, result) {
+      if (err) {
+        return self.callback(err, self._build_result(err));
+      }
+
+      self.emit('drain');
+    });
   }
 
-  return true;
+  return false;
```

Comment:
changing this to false prevents the caller from writing more data to the stream until we emit a `drain` event.